### PR TITLE
[Bigtable] Added StreamController cancel in readRowsAsync docs

### LINKED ISSUE #4229: bigtable: add docs demonstrating stream cancellation
docs for readRowsAsync should give an example of how to cancel the stream using the StreamController

### PR DESCRIPTION
**Pull request by rahulKQL**
Fixes #4229

This change contains a simple example of how to cancel a stream received from `BigtableDataClient#readRowsAsync`.

### CODE REVIEW

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 1]
```diff
@@ -540,29 +540,45 @@ public Row readRow(String tableId, ByteString rowKey, @Nullable Filter filter) {
    * <p>Sample code:
    *
    * <pre>{@code
+   * class ClientObserver implements ResponseObserver<Row> {
```
Comment (igorbernstein2) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: Please keep this as an anonymous inner class, something like:

```java

   *   bigtableDataClient.readRowsAsync(query, new ResponseObserver<Row>() {
   *     StreamController controller;
   *     int count = 0;
   *
   *     public void onStart(StreamController controller) {
   *         this.controller = controller;
   *     }
   *     public void onResponse(Row row) {
   *        if (++count > 10) {
  *            controller.cancel();
   *           return;
  *        }
   *       // Do something with Row
   *     }
   *     public void onError(Throwable t) {
   *       if (t instanceof NotFoundException) {
   *         System.out.println("Tried to read a non-existent table");
   *       } else {
   *         t.printStackTrace();
   *       }
   *     }
   *     public void onComplete() {
   *       // Handle stream completion
   *     }
   *   });

```

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java [hunk 2]
```diff
@@ -540,29 +540,45 @@ public Row readRow(String tableId, ByteString rowKey, @Nullable Filter filter) {
    * <p>Sample code:
```
Comment (igorbernstein2) [inline google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: Please add an example above as well:

```java
   ServerStream<Row> stream = ...
   int count = 0;

  for(Row row : ...) {
     if (++count > 10) {
       stream.cancel();
       break;
    }
   // Do something with row
...
  }

```

**Comment by rahulKQL** [comment]
@igorbernstein2 PTAL, Also please let me know if this sample code needs to be more concrete/realistic.

File: google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java (hunk 1 above)
Comment (rahulKQL) [reply google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java]: Have updated the sample code to include the described approach. Please have a fresh look

**Comment by rahulKQL** [comment]
@igorbernstein2
I have addressed both feedback comments. PTAL and merge if it looks good to you.

**Pull request merged**